src/lib.rs

```diff
@@ -14,6 +14,39 @@
 //! * To construct a pool, you must supply a [resolver::Resolver] and
 //!   a [backend::Connector]. These are interfaces which specify "how to find
 //!   backends" and "how to create connections to a backend", respectively.
+//!
+//! # DTrace probes
+//!
+//! qorb contains a number of DTrace USDT probes, which fire as the qorb pool
+//! manages its connections. These probes provide visibility into when qorb
+//! initiates connections; checks health; and vends out claims from a pool. The
+//! full list of probes is:
+//!
+//! - `claim-start`: Fires before attempting to make take a claim from the pool.
+//! - `claim-done`: Fires before returning a successful claim to the client.
+//! - `claim-failed`: Fires on failure to take a claim from the pool.
+//! - `connect-start`: Fires before attempting a connection to a backend.
+//! - `connect-done`: Fires after successfully connecting to a backend.
+//! - `connect-failed`: Fires after failing to connect to a backend.
+//! - `handle-claimed`: Fires after claiming a handle from the pool, before
+//!   returning it to the client.
+//! - `handle-recycled`: Fires when a handle is returned to the pool, after it
+//!   is dropped.
+//! - `health-check-start`: Fires when a health check for a backend starts.
+//! - `health-check-done`: Fires when a health check for a backend completes
+//!   successfully.
+//! - `health-check-failed`: Fires when a health check for a backend fails.
+//! - `recycle-start`: Fires before attempting to recycle a connection.
+//! - `recycle-done`: Fires after successsfully recycling a connection.
+//! - `recycle-failed`: Fires when failing to recycle a connection.
+//!
+//! The existence of the probes is behind the `"probes"` feature, which is
+//! enabled by default. Probes are zero-cost unless they are explicitly enabled,
+//! by tracing the program with the `dtrace(1)` command-line tool.
+//!
+//! Also note that consumers of the `qorb` pool need to call
+//! [`usdt::register_probes`] to register the probes with the OS. `qorb` does
+//! not do so automatically.
```

Comment:
IMO it would be really great to have something in `examples` that demos how to do this, and how to see the probes with dtrace - but that doesn't need to hold up this PR

Reply:
Sure that sounds good! Did you have thoughts on automatically registering the probes? It's very nice, but also fallible.